frontends/web/src/components/headerssync/headerssync.test.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { render } from '@testing-library/react';
+import { HeadersSync } from './headerssync';
+import { TStatus } from '../../api/coins';
+import * as apiHooks from '../../hooks/api';
+import * as utilHooks from '../../hooks/utils';
+
+const useSubscribeSpy = jest.spyOn(apiHooks, 'useSubscribe');
+const useMountedRefSpy = jest.spyOn(utilHooks, 'useMountedRef');
+
+
+describe('components/headerssync/headerssync', () => {
+  beforeEach(() => {
+    useMountedRefSpy.mockReturnValueOnce({ current: true });
+  });
+
+  it('renders null when subscribe value of null', () => {
+    useSubscribeSpy.mockReturnValueOnce(null);
+
+    const { container } = render(<HeadersSync coinCode="btc" />);
+    expect(container.firstChild).toBeNull();
```

Comment:
noob q: I'd expect container to be null, because headersync componentn returns null if there is no status, but container returns div here .. what am I missing?

Reply:
Yeah, in first glance it really could look like that. The thing is `render` wraps a `div` (which is this `container`)  on whatever is being `render`ed.

So the `firstChild` is the actual element (`HeaderSync` in this case).

_"Container: By default, React Testing Library will create a div and append that div to the document.body and this is where your React component will be rendered"_ - https://testing-library.com/docs/react-testing-library/api/#container